### PR TITLE
[RM Ch3] SmartSwitches

### PR DESCRIPTION
**Pull request by petorre**
Text from @qabpea .
Towards fixing #2051 .

### CODE REVIEW

File: doc/ref_model/chapters/chapter03.md [hunk 1]
```diff
@@ -289,3 +290,16 @@ The terms Container Infrastructure Service Instance and Container Infrastructure
 
 <a name="3.8.3"></a>
 ### 3.8.3 Decoupling applications from infrastructure with Hardware Acceleration
+
+
+<a name="3.8.8"></a>
+### 3.8.8 SmartSwitches
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter03.md:296]: Why the camel case?

**Review by CsatariGergely — CHANGES_REQUESTED** [review]
Can you please also describe what is the relationship of smart switches to RA2?

File: doc/ref_model/chapters/chapter03.md (hunk 1 above)
Comment (petorre) [reply doc/ref_model/chapters/chapter03.md]: > 
> 
> Why the camel case?

Changed. Thanks

**Comment by petorre** [comment]
> Can you please also describe what is the relationship of smart switches to RA2?

@qabpea : Maybe you could write this in one or two very concise sentences?
@CsatariGergely : This might need more work which would make it target for next release.

**Review by walterkozlowski — APPROVED** [review]

**Comment by walterkozlowski** [comment]
@TFredberg, @sureshkrishnan, @qabpea - please review and approve if you can

File: doc/ref_model/chapters/chapter03.md [hunk 2]
```diff
@@ -289,3 +290,16 @@ The terms Container Infrastructure Service Instance and Container Infrastructure
 
 <a name="3.8.3"></a>
 ### 3.8.3 Decoupling applications from infrastructure with Hardware Acceleration
+
+
+<a name="3.8.8"></a>
+### 3.8.8 Smart Switches
+
+Smart Swiches can be broadly categorized into Configurable Switches and Programable Switches.
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter03.md:298]:
```suggestion
Smart Switches can be broadly categorized into Configurable Switches and Programmable Switches.
```

File: doc/ref_model/chapters/chapter03.md [hunk 3]
```diff
@@ -289,3 +290,16 @@ The terms Container Infrastructure Service Instance and Container Infrastructure
...
+
+Configurable Smart Switches run generic “smart” configurable network operating system offering full range of network functionality and are flexible enough to support most network solutions. The most common such network operating system is Linux-based [SONiC](https://azure.github.io/SONiC/) allowing hardware and software disaggregation by running on switches from multiple switch vendors with different types of merchant fixed-function ASICs. Still, SONiC today cannot implement new type of data plane functionality or patch/modify/correct an ASIC, which is type of support is offered by programable smart switches.
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter03.md:300]:
```suggestion
Configurable Smart Switches run generic “smart” configurable network operating system offering full range of network functionality and are flexible enough to support most network solutions. The most common such network operating system is Linux-based [SONiC](https://azure.github.io/SONiC/) allowing hardware and software disaggregation by running on switches from multiple switch vendors with different types of vendor fixed-function ASICs. Still, SONiC today cannot implement new type of data plane functionality or patch/modify/correct an ASIC, which is the type of support offered by programmable smart switches.
```

File: doc/ref_model/chapters/chapter03.md [hunk 4]
```diff
@@ -289,3 +290,16 @@ The terms Container Infrastructure Service Instance and Container Infrastructure
...
+
+Programable Smart Switches make it possible to quickly support new or correct/modify existing protocols and network functions, allow end customers to implement network functions, and to only implement and load functionality that is needed. Such switches contain one or more programable switch ASICs of the same or different types. The two most used programming languages are [P4](https://p4.org/) and [NPL](https://nplang.org/), and both can be used with vendor-specific toolchains to program their switch ASICs and/or FPGAs. Open Networking Foundation [Stratum](https://opennetworking.org/stratum/) is an example of network operating system that offers generic life cycle management control services for the P4 components and a management API. The control API for the individual network functions are not part of the Stratum APIs.
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter03.md:302]:
```suggestion
Programmable Smart Switches make it possible to quickly support new or correct/modify existing protocols and network functions, allow end customers to implement network functions, and to only implement and load functionality that is needed. Such switches contain one or more programmable switch ASICs of the same or different types. The two most used programming languages are [P4](https://p4.org/) and [NPL](https://nplang.org/), and both can be used with vendor-specific toolchains to program their switch ASICs and/or FPGAs. Open Networking Foundation [Stratum](https://opennetworking.org/stratum/) is an example of network operating system that offers generic life cycle management control services for the P4 components and a management API. The control API for the individual network functions are not part of the Stratum APIs.
```

File: doc/ref_model/chapters/chapter03.md [hunk 5]
```diff
@@ -289,3 +290,16 @@ The terms Container Infrastructure Service Instance and Container Infrastructure
...
+
+Based on Smart Switches, for fully integrated edge and fabric products exist other solutions from vendors like Arista, Cisco or Kaloom.
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter03.md:304]: there is something either missing or should be reworded
```suggestion
Based on Smart Switches, products exist for fully integrated edge and fabric solutions from vendors like Arista, Cisco or Kaloom.
```

**Comment by petorre** [comment]
Resolved conflict.

File: doc/ref_model/chapters/chapter03.md [hunk 6]
```diff
@@ -345,18 +345,24 @@ Preferably, Application or Infrastructure acceleration can take benefit from und
 - For O-RAN Network functions: O-RAN Acceleration Abstraction Layer Interface.
 
 
-<a name="3.8.3"></a>
-### 3.8.3 Decoupling applications from infrastructure with Hardware Acceleration
+
+<a name="3.8.4"></a>
+### 3.8.4 CPU Instructions
+
+CPU architecture often includes instructions and execution blocks for most common compute-heavy algorithms like block cypher (example AES-NI), Random Number Generator or vector instructions. Those are normally consumed in software infrastructure or application by using enabled software libraries that will run faster when instructions are available in hardware and slower when instructions are not available in hardware, so other more general CPU instructions are used. CPU instructions don’t need to be activated or life-cycle-managed. Finding such compute nodes during scheduling workloads can be done by application control/orchestrator using OpenStack Nova filters or Kubernetes Node Feature Discovery labels, or directly from the Hardware Management layer.
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter03.md:352]:
```suggestion
The CPU architecture often includes instructions and execution blocks for most common compute-heavy algorithms like block cypher (example AES-NI), Random Number Generator or vector instructions. These functions are normally consumed in infrastructure software or applications by using enabled software libraries that run faster when custom CPU instructions for the execution of such functions are available in hardware and slower when these specific instructions are not available in hardware as only the general CPU instructions are used. Custom CPU instructions don’t need to be activated or life-cycle-managed. When scheduling workloads, compute nodes with such custom CPU instructions can be found by applications or an orchestrator using OpenStack Nova filters or Kubernetes Node Feature Discovery labels, or directly from the Hardware Management layer.
```

**Review by TFredberg — APPROVED** [review]

**Review by pgoyal01 — APPROVED** [review]

**Comment by collivier** [comment]
@CsatariGergely @petorre : is the conflict resolved? It's well reviewed but I read "This might need more work which would make it target for next release".

**Comment by petorre** [comment]
> @CsatariGergely @petorre : is the conflict resolved? It's well reviewed but I read "This might need more work which would make it target for next release".

@collivier : 
I resolved the conflicts.
Current text was prepared for Elbrus.
That comment meant that for more ideas we run out time to implement so those would go to next release. On February vF2F we have session for to discuss those.

**Pull request merged**